### PR TITLE
Use MP_SIZEOF_BITS for MP_WARRAY definition

### PR DESCRIPTION
**Pull request by nijtmans**
Also fix depreciation messages and remove unnecessary includes.

Just some random improvements, not worth to split in 3 different pull requests.

Inspired by #327

### CODE REVIEW

**Comment by minad** [comment]
mp_sizeof_bits cannot be used in tommath.h since it is private

**Comment by nijtmans** [comment]
> mp_sizeof_bits cannot be used in tommath.h since it is private

Indeed. Fixed now.

In addition, I think MP_SIZEOF_BITS() should return an int, not size_t, since it is used to compare against other integers, generally. Follow-up commits demonstrate more simplifications doing that.

**Comment by minad** [comment]
I think we should rather use size_t at more places instead of int. sizeof_bits should return size_t like sizeof.

**Comment by nijtmans** [comment]
> I think we should rather use size_t at more places instead of int. sizeof_bits should return size_t like sizeof.

If MP_SIZEOF_BITS was a public macro, I would agree with you on that. But MP_SIZEOF_BITS() is used internally, usually comparing with mp_int->used or mp_int->alloc, which are integers. That results in type-casts in the code, that could better be handled in MP_SIZEOF_BITS. This branch is meant to prove that: showing that changing MP_SIZEOF_BITS results in less type-casts in the code. Making it more readable.

But  ....   if you are fond of type-casts ....    ;-)

File: tommath.h [hunk 1]
```diff
@@ -517,7 +517,7 @@ mp_err mp_lcm(const mp_int *a, const mp_int *b, mp_int *c) MP_WUR;
  */
 mp_err mp_root_u32(const mp_int *a, uint32_t b, mp_int *c) MP_WUR;
 MP_DEPRECATED(mp_root_u32) mp_err mp_n_root(const mp_int *a, mp_digit b, mp_int *c) MP_WUR;
-MP_DEPRECATED(mp_n_root_ex) mp_err mp_n_root_ex(const mp_int *a, mp_digit b, mp_int *c, int fast) MP_WUR;
+MP_DEPRECATED(mp_root_u32) mp_err mp_n_root_ex(const mp_int *a, mp_digit b, mp_int *c, int fast) MP_WUR;
```
Comment (sjaeckel) [inline tommath.h]: can you please split this and the other `MP_DEPRECATED`-related change out into a separate PR

**Comment by sjaeckel** [comment]
> ... sizeof_bits should return size_t like sizeof.

I think so too

> If MP_SIZEOF_BITS was a public macro, I would agree with you on that. But MP_SIZEOF_BITS() is used internally, usually comparing with mp_int->used or mp_int->alloc, which are integers

so instead of making sizeof_bits return the wrong type we will change these types to size_t after 1.2.0 is released. This should make both of you happy and @karel-m unfortunately a bit sadder ... :-\

**Comment by nijtmans** [comment]
If there are plans to change mp_int->used and mp_int->alloc from int to size_t, yes that would make me happy (although it's binary incompatible on 64-bit platforms, so that cannot be done in 1.x)

But - anyway -   I'll close this one, if there are such plans, this PR makes no sense.

**Pull request closed without merging**